src/CarbonAware.DataSources/CarbonAware.DataSources.WattTime/test/WattTimeDataSourceTests.cs

```diff
@@ -247,6 +247,52 @@ public void GetCurrentCarbonIntensityForecastAsync_ThrowsWhenTooFewDatapointsRet
         Assert.ThrowsAsync<WattTimeClientException>(async () => await this.DataSource.GetCurrentCarbonIntensityForecastAsync(location));
     }
 
+    [TestCase("2022-01-01T00:00:00Z", "2022-01-01T00:00:00Z", TestName = "GetCarbonIntensityForecastAsync returns expected value 2022-01-01T00:00:00Z round to floor")]
+    [TestCase("2021-05-01T00:03:20Z", "2022-05-01T00:00:00Z", TestName = "GetCarbonIntensityForecastAsync returns expected value 2022-05-01T00:00:00Z round to floor")]
+    [TestCase("2022-01-01T00:07:00Z", "2022-01-01T00:05:00Z", TestName = "GetCarbonIntensityForecastAsync returns expected value 2022-01-01T00:05:00Z round to floor")]
+    public async Task GetCarbonIntensityForecastAsync_RequiredAtRounded(string requested, string expected)
+    {
+        // Arrange
+        var location = new Location() { RegionName = "eastus", LocationType = LocationType.CloudProvider, CloudProvider = CloudProvider.Azure };
+        var balancingAuthority = new BalancingAuthority() { Abbreviation = "BA" };
+        var requestedAt = DateTimeOffset.Parse(requested);
+        var expectedAt = DateTimeOffset.Parse(expected);
+        var startPoint = DateTimeOffset.Parse("2022-01-01T00:00:00Z");
+
+        var emissionData = new List<GridEmissionDataPoint>()
+        {
+            new GridEmissionDataPoint()
+            {
+                BalancingAuthorityAbbreviation = balancingAuthority.Abbreviation,
+                PointTime = startPoint,
+                Value = 10,
+            },
+            new GridEmissionDataPoint()
+            {
+                BalancingAuthorityAbbreviation = balancingAuthority.Abbreviation,
+                PointTime = startPoint + TimeSpan.FromMinutes(5),
+                Value = 10,
+            },
+        };
+        var forecast = new Forecast()
+        {
+            GeneratedAt = startPoint,
+            ForecastData = emissionData
+        };
+
+        SetupBalancingAuthority(balancingAuthority, location);
+        this.WattTimeClient.Setup(w => w.GetForecastOnDateAsync(balancingAuthority, expectedAt)
+                ).ReturnsAsync(() => forecast);
+
+        // Act
+        var result = await this.DataSource.GetCarbonIntensityForecastAsync(location, requestedAt);
+
+        // Assert
+        Assert.IsNotNull(result);
+        Assert.AreEqual(result.RequestedAt, expectedAt);
```

Comment:
```suggestion
        Assert.AreEqual(result.RequestedAt, requestedAt);
        this.WattTimeClient.Verify(w => w.GetForecastOnDateAsync(It.IsAny<BalancingAuthority>(), It.Is<DateTimeOffset>(r => r == expectedAt)), Times.Once);
```

This should test the new expectations